File: src/main/java/factory_method/product/SuperBus.java
package factory_method.product;

import factory_method.product.part.Engine;
import factory_method.product.part.Tire;

public class SuperBus extends Vehicle {
    public static final String NAME = "Super Bus";

    public SuperBus() {
        name = NAME;
        engine = Engine.HYBRID;
        tire = Tire.NORMAL;
    }
}

File: src/main/kotlin/factory_method/product/SuperBus.kt
package factory_method.product

import factory_method.product.part.Engine
import factory_method.product.part.Tire

class SuperBus : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.HYBRID
    override var tire: Tire = Tire.NORMAL

    companion object {
        const val NAME: String = "Super Bus"
    }
}